libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewSubscriber.java

```diff
@@ -1,24 +1,34 @@
 package com.mapbox.services.android.navigation.ui.v5;
 
+import android.arch.lifecycle.Lifecycle;
+import android.arch.lifecycle.LifecycleObserver;
 import android.arch.lifecycle.LifecycleOwner;
 import android.arch.lifecycle.Observer;
+import android.arch.lifecycle.OnLifecycleEvent;
 import android.location.Location;
 import android.support.annotation.Nullable;
 
 import com.mapbox.api.directions.v5.models.DirectionsRoute;
 import com.mapbox.geojson.Point;
 
-class NavigationViewSubscriber {
+class NavigationViewSubscriber implements LifecycleObserver {
 
-  private NavigationPresenter navigationPresenter;
+  private final NavigationPresenter navigationPresenter;
+  private final NavigationViewModel navigationViewModel;
+  private final LifecycleOwner lifecycleOwner;
 
-  NavigationViewSubscriber(NavigationPresenter navigationPresenter) {
+  NavigationViewSubscriber(final LifecycleOwner owner,
+                           final NavigationViewModel navigationViewModel,
```

Comment:
NIT no need for new lines, also could we rearrange the properties to match constructor for better readability?

Reply:
I'm agree with you, Done